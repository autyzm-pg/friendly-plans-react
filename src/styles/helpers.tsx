import { ViewStyle } from 'react-native';
import { palette } from './palette';

export const getElevation = (elevation: number): ViewStyle => ({
  elevation,
  shadowColor: palette.shadow,
  shadowOffset: {
    width: 0,
    height: Math.round(1 + elevation / 4),
  },
  shadowOpacity: 0.17 + elevation / 100,
  shadowRadius: 1 + elevation / 2,
});
