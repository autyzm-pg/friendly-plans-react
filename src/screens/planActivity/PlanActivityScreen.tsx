import isEmpty from 'lodash.isempty';
import React from 'react';
import { StyleSheet, View } from 'react-native';
import { NavigationInjectedProps } from 'react-navigation';

import { i18n } from 'locale';
import { Plan } from 'models';
import { getElevation, palette } from 'styles';
import { FixedCreatePlanItemButton } from './FixedCreatePlanItemButton';
import { PlanForm, PlanFormData } from './PlanForm';
import { TaskTable } from './TaskTable';
import { TaskTableHeader } from './TaskTableHeader';

interface State {
  rowList: number[];
  plan: Plan;
}

export class PlanActivityScreen extends React.PureComponent<NavigationInjectedProps, State> {
  static navigationOptions = {
    title: i18n.t('planList:viewTitle'),
  };

  state: State = {
    rowList: [],
    plan: this.props.navigation.getParam('plan'),
  };

  handleAddRow = () => {
    this.setState({ rowList: [...this.state.rowList, this.state.rowList.length] });
  };

  createPlan = async (name: string) => {
    const { id } = this.props.navigation.getParam('student');

    const plan = await Plan.createPlan(id, name);

    this.setState({ plan });
  };

  updatePlan = async (name: string) => {
    await this.state.plan.update({
      name,
    });

    this.setState({ plan: { ...this.state.plan, name } });
  };

  onSubmit = ({ planInput }: PlanFormData) =>
    this.state.plan ? this.updatePlan(planInput) : this.createPlan(planInput);

  navigateToCreatePlanItem = async (itemType: string) => {
    const student = this.props.navigation.getParam('student');
    const plan = this.state.plan;

    this.props.navigation.navigate('PlanItemTask', {
      student,
      plan,
    });
  };

  render() {
    const { plan, rowList } = this.state;

    return (
      <>
        <View style={styles.headerContainer}>
          <PlanForm onSubmit={this.onSubmit} plan={plan} />
          {!isEmpty(rowList) && <TaskTableHeader />}
        </View>
        <TaskTable rowList={rowList} />
        {plan && <FixedCreatePlanItemButton onPress={this.navigateToCreatePlanItem} />}
      </>
    );
  }
}

const styles = StyleSheet.create({
  headerContainer: {
    ...getElevation(5),
    backgroundColor: palette.background,
  },
});
