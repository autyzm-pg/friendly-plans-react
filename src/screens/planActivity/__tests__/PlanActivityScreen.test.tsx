import { shallow } from 'enzyme';
import React from 'react';

import { navigationMock } from 'helpers';
import { PlanActivityScreen } from '../PlanActivityScreen';

jest.mock('models/ModelSubscriber', () => ({
  ModelSubscriber: jest.fn().mockImplementation(() => ({
    subscribeCollectionUpdates: jest.fn(),
    unsubscribeCollectionUpdates: jest.fn(),
  })),
}));

describe('PlanActivityScreen', () => {
  const props = {
    navigation: navigationMock,
  };

  it('renders correctly', () => {
    const planActivityScreen = shallow(<PlanActivityScreen {...props} />);

    expect(planActivityScreen).toMatchSnapshot();
  });
});
