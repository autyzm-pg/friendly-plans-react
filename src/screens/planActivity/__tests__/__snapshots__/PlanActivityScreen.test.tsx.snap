// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanActivityScreen renders correctly 1`] = `
<Fragment>
  <FullScreenTemplate>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "elevation": 5,
          "shadowColor": "rgba(0, 0, 0, 0.7)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.22000000000000003,
          "shadowRadius": 3.5,
        }
      }
    >
      <Component
        onSubmit={[Function]}
        onValidate={[Function]}
        plan="plan"
        playDisabled={true}
        shuffleDisabled={true}
      />
    </View>
    <Component
      planItemList={Array []}
    />
  </FullScreenTemplate>
  <FixedCreatePlanItemButton
    onPress={[Function]}
  />
</Fragment>
`;
